src/main/java/com/internet/mixshop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,175 @@
+import com.internet.mixshop.model.Order;
+import com.internet.mixshop.model.Product;
+import com.internet.mixshop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class OrderDaoJdbcImpl implements OrderDao {
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        String query = "SELECT * FROM orders WHERE user_id = ? AND is_deleted = false";
+        List<Order> userOrders = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                userOrders.add(buildOrder(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get orders by user: "
+                    + userId, e);
+        }
+        for (Order order : userOrders) {
+            order.setProducts(getProductsListByOrderId(order.getId()));
+        }
+        return userOrders;
+    }
+
+    @Override
+    public Order create(Order order) {
+        String query = "INSERT INTO orders(user_id) VALUES (?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query,
+                    Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            while (resultSet.next()) {
+                order.setId(resultSet.getLong(1));
+            }
+            addProductsToOrder(order.getProducts(), order.getId());
+            return order;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Order with id " + order.getId()
+                    + " is not added!", e);
+        }
+    }
+
+    @Override
+    public Optional<Order> getById(Long id) {
+        String query = "SELECT * FROM orders WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Order order = getOrder(resultSet, connection);
+                return Optional.ofNullable(order);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get order with id = " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Order update(Order order) {
```

Comment:
In this method you should also update the products in the order (first delete them all, and then add new products from your object)

Reply:
done